### PR TITLE
Chef client running as a windows service should have a configurable timeout

### PR DESCRIPTION
**Pull request by jdmundrawala**
Solves https://github.com/chef/chef/issues/2985

cc @btm @smurawski @ksubrama @adamedx

### CODE REVIEW

File: lib/chef/config.rb [hunk 1]
```diff
@@ -643,6 +643,12 @@ def self.windows_home_path
     default :normal_attribute_whitelist, nil
     default :override_attribute_whitelist, nil
 
+    config_context :windows_service do
+      # Set `watchdog_timeout` to the number of seconds to wait for a chef-client run
+      # to finish
+      default :watchdog_timeout, 20 * 60
```
Comment (danielsdeleo) [inline lib/chef/config.rb]: This seems pretty low to me. Is there any risk with setting to to something like 2 hours?

Comment (jdmundrawala) [inline lib/chef/config.rb]: I think 2 hours would be ok. @adamedx thoughts?

Comment (adamedx) [inline lib/chef/config.rb]: I'm just a little concerned about our default frequency -- chef-client is set to run every 30 minutes, correct? As long as we're comfortable with the way chef-client handles jobs that exceed the default execution period of 30 minutes, I'm cool with 2 hours. Reality is that people do things like SQL Server installations + .net stack that can actually take significant amounts of time, so erring on the longer side if it is safe is a good recommendation.

**Comment by danielsdeleo** [comment]
:+1: 

For those interested, we confirmed that the windows service app just loops so it won't create stacking chef-client processes: https://github.com/chef/chef/blob/fb95d688b483700d0c62b6a6960cd2b614121a7e/lib/chef/application/windows_service.rb#L80

**Comment by adamedx** [comment]
Thanks for validating that! :+1:

**Pull request merged**